### PR TITLE
code cleanup

### PR DESCRIPTION
**Pull request by chavacava**
Little code refactor with the goal of making code more go idiomatic and easy to understand.

### CODE REVIEW

File: rule/max-public-structs.go [hunk 1]
```diff
@@ -19,20 +19,22 @@ const defaultMaxPublicStructs = 5
...
+	}
 
-		max, ok := arguments[0].(int64) // Alt. non panicking version
-		if !ok {
-			panic(`invalid value passed as argument number to the "max-public-structs" rule`)
-		}
-		r.max = max
+	checkNumberOfArguments(1, arguments, r.Name())
+
+	max, ok := arguments[0].(int64) // Alt. non panicking version
```
Comment (denisvmedia) [inline rule/max-public-structs.go:33]: Note, max here will override a built-in.

Comment (chavacava) [reply rule/max-public-structs.go:33]: right, I've created this branch before the last merged PR #1052, I will rebase once I've refactored all rules

**Comment by denisvmedia** [comment]
There is some issue with 1.21 tests, but other than that I really like the way how you refactored the code.

**Comment by chavacava** [comment]
> There is some issue with 1.21 tests, but other than that I really like the way how you refactored the code.

Testing step fails before actually running the test (some problem at the preparation of the environment) I'll check what is wrong

**Pull request closed without merging**